### ISSUE TITLE
Astro.cookies.set() not adding cookie in vercel adapter

### ISSUE THREAD

**Issue by karolwydmuch** [withastro/astro#9831]
### Astro Info

```block
Astro                    v4.2.4
Node                     v18.19.0
System                   macOS (arm64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/vercel/serverless
Integrations             @astrojs/svelte
```

### Describe the Bug

I deployed app that works correctly on localhost to vercel and I can't set cookies during login request (neither GET and POST).

1. user enters login page
2. user write correct login and password
3. on login/index.astro I set cookie like on example below
4. no set cookie in the header
5. on the middleware I'm trying to read this cookie but no value

`src/pages/login/index.astro`
```
---
Astro.cookies.set('test', 'am i working?', {
    httpOnly: true,
    sameSite: "lax",
    secure: true,
    path: "/"
})
---
```

`src/middleware/index.ts`
part of:
`const token = cookies.get('token').value` - throws error - no value

example request that should set this cookie
<img width="250" alt="Screenshot 2024-01-25 at 22 01 00" src="https://github.com/withastro/astro/assets/85114732/e7f1d1dd-89b5-48d5-9d1f-dfbe42c98f25">

### What's the expected result?

cookies just works (like on localhost). Maybe it's my bad but please help :D

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-cypmwy

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by stuartwk** [comment]
I'm experience the same thing. Rolling back to previous versions of dependencies worked for me.

**Comment by florian-lefebvre** [comment]
A fix is ready and will be available soon (in 7.0.3). I'll link the related PR to this issue so that you know when to expect the patch (when it's merged)
